src/simulators.jl

```diff
@@ -38,7 +38,6 @@ plot(t, s.y(sol, t)[:], lab="Open loop step response")
 """
 function Simulator(P::AbstractStateSpace, u::F = (x,t) -> 0) where F
     @assert iscontinuous(P) "Simulator only supports continuous-time system. See function `lsim` for simulation of discrete-time systems."
-    @assert all(P.D .== 0) "Can not simulate systems with direct term D != 0"
     f = (dx,x,p,t) -> dx .= P.A*x .+ P.B*u(x,t)
     y(x,t) = P.C*x #.+ P.D*u(x,t)
```

Comment:
```suggestion
    y(x,t) = P.C*x .+ P.D*u(x,t)
```
I guess you have to add the contribution of the input to the output as well

Reply:
So this is another thing I mentioned in #397, it works as is since `lsim` does not use this `y` function to calculate `y` but instead does it by itself. It feels like `lsim` should either use this (after fixing it) or maybe remove `y` here. I was not sure of the preferred way so I started with only the assertion to test if it ran in the CI pipeline.

I'm not sure I understand the benefit of the `Simulator` type at all. Since it is only used in `lsim` it seems like it would be easier and less code to just write out the simulation there? But either solution is fine for me, wanted some input before I started to make more stuff.

Reply:
It should probably still be there, since we put the `y` function in the `Simulator` struct, so the user might call it later and then expects it to contain `Du` if `D != 0`. There is one more method of `y` added below that is also affected.

Reply:
Agreed, not sure why I didn't accept the suggestion before.